hudi-flink/src/main/java/org/apache/hudi/sink/bootstrap/BootstrapFunction.java

```diff
@@ -193,22 +205,22 @@ protected void loadRecords(String partitionPath, Collector<O> out) throws Except
           }
 
           for (HoodieKey hoodieKey : hoodieKeys) {
-            out.collect((O) new IndexRecord(generateHoodieRecord(hoodieKey, fileSlice)));
+            output.collect(new StreamRecord(new IndexRecord(generateHoodieRecord(hoodieKey, fileSlice))));
           }
         });
 
         // load avro log records
         List<String> logPaths = fileSlice.getLogFiles()
-                // filter out crushed files
-                .filter(logFile -> isValidFile(logFile.getFileStatus()))
-                .map(logFile -> logFile.getPath().toString())
-                .collect(toList());
+            // filter out crushed files
+            .filter(logFile -> isValidFile(logFile.getFileStatus()))
+            .map(logFile -> logFile.getPath().toString())
```

Comment:
We can promote the code to load the keys incrementally, say, the new keys since last commit state instant.

Reply:
We can do that in following PR.